### DISCUSSION TITLE
Common syntax for C# 12 collection literals

### DISCUSSION THREAD

**Discussion by lkomanetz** [dotnet/csharplang#7347] [Language Ideas] [answered]
Is it possible to have the new C# 12 collection literal syntax use curly braces `{}` instead of square braces `[]`?  The only reason that comes to mind is so syntax can stay consistent.  Right now if I'm going to initialize a collection of numbers I would go
```cs
int[] numbers = new[] { 1, 2, 3 };
```
If I would use the new C# 12 feature I would go about the same thing using different symbols than the language has gotten me used to.  It wouldn't be much to change my brain but since the precedent has been set with curly braces it would be nice to keep them around for this feature too.
```cs
int[] numberArray = { 1, 2, 3 };
List<int> numberList = { 4, 5, 6 };
Span<int> numberSpan = { 8, 9, 10 };
int[] composed = { ..numberArray, ..numberList, 7, ..numberSpan };
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/issues/5354#issuecomment-1636384116

TL;DR:  curly braces are overloaded in the language as it is, it would not have been possible to have a literal syntax with a symmetric deconstruction syntax without ambiguity.

**Comment by CyrusNajmabadi** [comment]
Please have this discussion on the actual design discussion for collection literals.
